### PR TITLE
custom-set: Stub, order of methods, rename some methods

### PR DESCRIPTION
**Pull request by sshine**
For this PR I haven't written a test generator yet, as it was recommended in #241.

### CODE REVIEW

**Comment by sshine** [comment]
Should we avoid features like subroutine signatures in example solutions because they don't exist in Perl 5.18 and below?

I've intentionally omitted them from the stub to stay backwards compatible, but included them in the example solution because I consider them idiomatic within some style of modern Perl.

Perhaps examples should only be tested with later versions of Perl, since the student is not required to seek them out.

**Comment by mienaikage** [comment]
I would say it's best to avoid them in the examples, as I figure the primary intention of the examples is to ensure that the tests work properly across all versions. Signatures are nice, but are seemingly rarely used in Perl 5. Maybe it'd be worth adding documentation with a link to a list of perl features (e.g. https://perldoc.pl/feature) for users to explore?

**Comment by sshine** [comment]
> I figure the primary intention of the examples is to ensure that the tests work properly across all versions

OK. I'll remove them from the example.

> Signatures are nice, but are seemingly rarely used in Perl 5.

We use them at work all the time; that is, old code still hasn't got them, but when I refactor old subroutines that appear to be called in predictable ways, I always add them as a measure of safety, since signatures will error on the wrong number of arguments, which is a weak kind of type-safety.

File: exercises/custom-set/CustomSet.pm [hunk 1]
```diff
@@ -0,0 +1,77 @@
+package CustomSet;
+use strict;
+use warnings;
+use Exporter 'import';
```
Comment (mienaikage) [inline exercises/custom-set/CustomSet.pm]: We don't need the exporter for this as all the subs are class methods.

File: exercises/custom-set/CustomSet.pm [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+our @EXPORT_OK = qw(
+    new add remove is_empty
+    is_member size to_list
+    union intersect difference
+    is_disjoint is_equal is_subset
+);
+
+sub new {
+    my ($class, @members) = @_;
+    die "You need to implement this.";
```
Comment (mienaikage) [inline exercises/custom-set/CustomSet.pm]: Maybe we could use [the ellipsis statement](https://perldoc.pl/perlsyn#The-Ellipsis-Statement) instead? (perhaps commenting one with this link)

File: exercises/custom-set/CustomSet.pm (hunk 1 above)
Comment (sshine) [reply exercises/custom-set/CustomSet.pm]: Do we need `our @EXPORT_OK = ...` then?
```suggestion
```

File: exercises/custom-set/CustomSet.pm [hunk 3]
```diff
@@ -8,6 +8,9 @@ our @EXPORT_OK = qw(
     is_disjoint is_equal is_subset
 );
 
+# Replace `...` with a correct implementation.
+# https://perldoc.pl/perlsyn#The-Ellipsis-Statement
```
Comment (sshine) [inline exercises/custom-set/CustomSet.pm]: @mienaikage: Like so?

File: exercises/custom-set/.meta/solutions/CustomSet.pm [hunk 4]
```diff
@@ -1,91 +1,81 @@
 package CustomSet;
-
 use strict;
 use warnings;
+use Exporter 'import';
```
Comment (sshine) [inline exercises/custom-set/.meta/solutions/CustomSet.pm]:
```suggestion
```

File: exercises/custom-set/CustomSet.pm (hunk 1 above)
Comment (mienaikage) [reply exercises/custom-set/CustomSet.pm]: We don't need it. (Sorry for lack of clarity I meant to include `@EXPORT_OK` with the comment)

File: exercises/custom-set/CustomSet.pm (hunk 3 above)
Comment (mienaikage) [reply exercises/custom-set/CustomSet.pm]: Looks good!

**Comment by sshine** [comment]
I've refrained from adding generator support (#249).

That issue says you'd like to migrate tests to Test2 first.

I'd consider both out of the scope of this PR.

Let me know if the PR is missing anything.

**Comment by mienaikage** [comment]
Pretty much looks good to me, just need to take out the `@EXPORT_OK`

**Comment by sshine** [comment]
OK, didn't get that. They're removed now.

**Pull request merged**